site/src/components/TemplateLayout/TemplatePageHeader.tsx

```diff
@@ -80,6 +101,14 @@ export const TemplatePageHeader: FC<TemplatePageHeaderProps> = ({
                 onClick={deleteTemplate.openDeleteConfirmation}
               />
               <TemplateSettingsButton templateName={template.name} />
+              <Maybe
+                condition={
+                  templateVersionVariables &&
+                  templateVersionVariables.length > 0
+                }
```

Comment:
If we don't load the templateVersionVariables here, which I think we don't need, it is ok to have the button sending the user to a page with an empty message. Hide actions are usually not a good pattern, it is better to have them displayed and some nice information about why it is empty or disabled.